Web Quiz Engine (Kotlin)/task/src/engine/service/QuizService.kt

```diff
@@ -24,6 +28,21 @@ class QuizService(
             )
         }
 
+    fun deleteQuizById(quizId: Int): ResponseEntity<Unit> =
+        with(findQuizByIdOrThrow(quizId = quizId)) {
+            val user = SecurityContextHolder.getContext().authentication.principal as QuizUser
```

Comment:
You should think about creating a static accessor for `SecurityContextHolder.getContext().authentication.principal as QuizUser`.

It could be something like a package `util` with a static class `SecurityUtil` with a method `getUser()`.

Reply:
Ok I'm not saying I'm against though in general I tend to stay away from utils like patterns as they become a dumpster for everything that doesn't fit. However here we're two grown ups responsible adults so I'm on it.
Done

Reply:
You can name the package and class anything you like, the point is to make this logic available statically.

Reply:
Done leveraging kotlin `object` with a property
```kotlin
object SecurityUtil {
    val authenticatedUser =
        SecurityContextHolder.getContext().authentication.principal as QuizUser
```